cachito/workers/tasks/rubygems.py

```diff
@@ -25,8 +41,78 @@ def cleanup_rubygems_request(request_id):
     nexus.execute_script("rubygems_cleanup", payload)
 
 
+@app.task
+@runs_if_request_in_progress
+def fetch_rubygems_source(request_id, package_configs=None):
+    """
+    Resolve and fetch RubyGems dependencies for a given request.
+
+    :param int request_id: the Cachito request ID this is for
+    :param list package_configs: the list of optional package configurations submitted by the user
+    """
+    validate_rubygems_config()
+    bundle_dir: RequestBundleDir = RequestBundleDir(request_id)
+
+    log.info("Configuring Nexus for RubyGems for the request %d", request_id)
+    set_request_state(request_id, "in_progress", "Configuring Nexus for RubyGems")
+    rubygems_repo_name = get_rubygems_hosted_repo_name(request_id)
+    prepare_nexus_for_rubygems_request(rubygems_repo_name)
+
+    log.info("Fetching dependencies for request %d", request_id)
+    package_configs = package_configs or [{}]
+    packages_data = []
+    for pkg_cfg in package_configs:
+        pkg_path = pkg_cfg.get("path", ".")
```

Comment:
IIUC, this is submitted by user, thus shall we do some sort of validation to prevent any potential file path attack?

https://owasp.org/www-community/attacks/Path_Traversal

Reply:
shouldn't be `normpath` also used here?

Reply:
I copy pasted it from pip implementation, so I didn't think much about it, but I'll look into it

Reply:
The paths are relative
https://github.com/containerbuildsystem/cachito/blob/fcde3cbccd21a17eebb3543ec9c3a3314bd76245/cachito/web/models.py#L142
There are no symlinks that lead outside the repo
https://github.com/containerbuildsystem/cachito/blob/fcde3cbccd21a17eebb3543ec9c3a3314bd76245/cachito/workers/tasks/general.py#L80

Reply:
+1 to use `normpath`, but IMO the extra validation is unnecessary

Reply:
I was thinking whether there was some validation already in place. So I suppose no other validation is necessary then?

Reply:
Ok, so as of right now, I only added normpath.